code/modules/antagonists/heretic/items/corrupted_organs.dm

```diff
@@ -0,0 +1,239 @@
+		LAZYADD(hallucinations, invisible_man)
+
+	if (LAZYLEN(hallucinations))
+		organ_owner.client.images |= hallucinations
+
+/obj/item/organ/internal/eyes/corrupt/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	if (!LAZYLEN(hallucinations))
+		return
+	organ_owner.client?.images -= hallucinations
+	QDEL_NULL(hallucinations)
+
+
+/// Sometimes speak in incomprehensible tongues
+/obj/item/organ/internal/tongue/corrupt
+	name = "corrupt tongue"
+	desc = "This one tells only lies."
+
+/obj/item/organ/internal/tongue/corrupt/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/corrupted_organ)
+	AddElement(/datum/element/noticable_organ, "The inside of %PRONOUN_Their mouth is full of stars.", BODY_ZONE_PRECISE_MOUTH)
+
+/obj/item/organ/internal/tongue/corrupt/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+	RegisterSignal(organ_owner, COMSIG_MOB_SAY, PROC_REF(on_spoken))
+
+/obj/item/organ/internal/tongue/corrupt/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	UnregisterSignal(organ_owner, COMSIG_MOB_SAY)
+
+/// When the mob speaks, sometimes put it in a different language
+/obj/item/organ/internal/tongue/corrupt/proc/on_spoken(mob/living/organ_owner, list/speech_args)
+	SIGNAL_HANDLER
+	if (organ_owner.has_reagent(/datum/reagent/water/holywater) || prob(60))
+		return
+	speech_args[SPEECH_LANGUAGE] = /datum/language/shadowtongue
+
+
+/// Randomly secretes alcohol or hallucinogens when you're drinking something
+/obj/item/organ/internal/liver/corrupt
+	name = "corrupt liver"
+	desc = "After what you've seen you could really go for a drink."
+	/// How much extra ingredients to add?
+	var/amount_added = 5
+	/// What extra ingredients can we add?
+	var/list/extra_ingredients = list(
+		/datum/reagent/consumable/ethanol/pina_olivada,
+		/datum/reagent/consumable/ethanol/rum,
+		/datum/reagent/consumable/ethanol/thirteenloko,
+		/datum/reagent/consumable/ethanol/vodka,
+		/datum/reagent/consumable/superlaughter,
+		/datum/reagent/drug/bath_salts,
+		/datum/reagent/drug/blastoff,
+		/datum/reagent/drug/happiness,
+		/datum/reagent/drug/mushroomhallucinogen,
+	)
+
+/obj/item/organ/internal/liver/corrupt/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/corrupted_organ)
+
+/obj/item/organ/internal/liver/corrupt/on_mob_insert(mob/living/carbon/organ_owner, special)
+	. = ..()
+	RegisterSignal(organ_owner, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_drank))
+
+/obj/item/organ/internal/liver/corrupt/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	UnregisterSignal(organ_owner, COMSIG_ATOM_EXPOSE_REAGENTS)
+
+/// If we drank something, add a little extra
+/obj/item/organ/internal/liver/corrupt/proc/on_drank(atom/source, list/reagents, datum/reagents/source_reagents, methods)
+	SIGNAL_HANDLER
+	if (!(methods & INGEST))
+		return
+	var/datum/reagents/extra_reagents = new()
+	extra_reagents.add_reagent(pick(extra_ingredients), amount_added)
```

Comment:
Similarly it feels kinda lame that the user has no way of knowing they're getting fucked up from the organ specifically, here. 
How about...
```suggestion
	extra_reagents.add_reagent(pick(extra_ingredients), amount_added)
	if(prob(20))
		to_chat(source, span_warning("As you feel the liquid travel down your throat, you feel like there's somehow more than what you drank. The sensation is intensely uncomfortable."))	
```